one_fm/hiring/doctype/onboard_employee/onboard_employee.js

```diff
@@ -152,48 +151,58 @@ var set_filters = function(frm) {
 };
 
 var create_custom_buttons = function(frm) {
-	if(frm.doc.applicant_attended && !frm.doc.work_contract){
-		cutom_btn_and_action(frm, 'create_work_contract', 'Work Contract');
-	}
-	if(frm.doc.electronic_signature_declaration_status != 1 && !frm.doc.declaration_of_electronic_signature){
-		cutom_btn_and_action(frm, 'create_declaration_of_electronic_signature', 'Electronic Signature Declaration');
-	}
-	if(frm.doc.work_contract_status == "Applicant Signed" && frm.doc.workflow_state == 'Declaration of Electronic Signature' && !frm.doc.duty_commencement){
-		cutom_btn_and_action(frm, 'create_duty_commencement', 'Duty Commencement');
-	}
-	if(frm.doc.duty_commencement_status == "Applicant Signed and Uploaded" && !frm.doc.employee){
-		cutom_btn_and_action(frm, 'create_employee', 'Employee');
-	}
-	if(frm.doc.employee && !frm.doc.employee_id){
-		cutom_btn_and_action(frm, 'create_employee_id', 'Employee ID Card');
-	}
-	if(frm.doc.employee && !frm.doc.user_created){
-		cutom_btn_and_action(frm, 'create_user_and_permissions', 'ERPNext User');
-	}
-	if(frm.doc.employee && frm.doc.user_created && !frm.doc.inform_and_send_enrolment_details_to_employee){
-		frm.add_custom_button(__('Inform and Send Enrolment Details to Employee'), function() {
-			frm.set_value('inform_and_send_enrolment_details_to_employee', true);
-			frm.save("Update");
-		}).addClass('btn-primary');
-	}
-	if(frm.doc.employee && !frm.doc.bank_account){
-		cutom_btn_and_action(frm, 'create_bank_account', 'Bank Account');
-	}
-	if(frm.doc.employee && !frm.doc.loan && frm.doc.net_loan_amount > 0){
-		frm.add_custom_button(__('Loan'), function() {
-			btn_create_loan_action(frm);
-		}, __('Create')).addClass('btn-primary');
-	}
-	if(frm.doc.employee && !frm.doc.mgrp && frm.doc.nationality == 'Kuwaiti'){
-		cutom_btn_and_action(frm, 'create_mgrp', 'MGRP');
-	}
-	if(frm.doc.employee && !frm.doc.pifss_form_103 && frm.doc.nationality == 'Kuwaiti'){
-		cutom_btn_and_action(frm, 'create_103_form', 'PIFSS Form 103');
-	}
-	if(frm.doc.employee && frm.doc.tools_needed_for_work && !frm.doc.request_for_material){
-		cutom_btn_and_action(frm, 'create_rfm_from_eo', 'Request for Material');
+	/*
+		Button for creating the Work Contract, Declaration of Electronic Signature, Duty Commencement, Bank Account
+		will comes in the below order asper the workflow in Onboard Employee
+		1. Work Contract  - Create if Applicant Attended the Orientation
+		2. Declaration of Electronic Signature - Create after the Work Contract only
```

Comment:
Hey @pjamsheer , I want to correct you here. The declaration is supposed to be done before the work Contract.

Reply:
@SalShai asper the Workflow given in the Story #180618303 and the flow described by Abdullah.
First work contract will signed by the applicant in a tablet as a digital signature and if he signed then only we go for creating a declaration of electronic signature for future use.
While we creating Electronic Signature Declaration, we need to set the electronic signature from the work contract and then we will have his/her actual signature on the print with the electronic signature